lib/eco/lspace_ext/lspace.py

```diff
@@ -0,0 +1,126 @@
+import os, tempfile, json, subprocess
+
+from incparser.astree import TextNode, BOS, EOS, ImageNode, FinishSymbol
+from grammar_parser.gparser import Terminal, MagicTerminal, IndentationTerminal, Nonterminal
+
+
+class CannotFindLSpaceRootError (Exception):
+    pass
+
+class CannotLaunchLSpaceError (Exception):
+    pass
+
+
+if 'LSPACE_ROOT' in os.environ:
+    LSPACE_ROOT = os.environ['LSPACE_ROOT']
+else:
+    LSPACE_ROOT = os.path.expanduser(os.path.join(
+        '~', 'kcl', 'lspace'))
+
+
+def _valid_lspace_root(lspace_root=None):
```

Comment:
I think @ltratt  and I agreed to only have one path pointing to external stuff to no confuse the user with multiple settings. In Eco's case we decided to only use the settings and not use environment vars at all.

Quote from https://github.com/softdevteam/eco/pull/65:
> I think it's probably best to remove the environment stuff, otherwise people might define the variable outside Eco, > then define it inside, and be confused as to which actually effects the outcome.